lib/models/blueprint.js

```diff
@@ -850,7 +850,9 @@ Blueprint.prototype._locals = function(options) {
   var sanitizedModuleName = moduleName.replace(/\//g, '-');
 
   return new Promise(function(resolve) {
-    resolve(this.locals(options));
+    return resolve(this._processPrompts(options));
+  }.bind(this)).then(function (prompts) {
```

Comment:
I prefer the constructor form, but maybe our promise subclass whats to add `Promise.invoke(context, method, [args])` ?